src/libraries/System.Net.Sockets/tests/FunctionalTests/CreateSocketTests.cs

```diff
@@ -233,5 +235,233 @@ public void Ctor_Netcoreapp_Success(AddressFamily addressFamily)
+            using var server = await listener.AcceptAsync();
+
+            using var client = new Socket(orig.SafeHandle);
+
+            Assert.True(client.Connected);
+            Assert.Equal(orig.AddressFamily, client.AddressFamily);
+            Assert.Equal(orig.SocketType, client.SocketType);
+            Assert.True(client.ProtocolType == orig.ProtocolType || client.ProtocolType == ProtocolType.Unknown, $"Expected: {protocolType} or Unknown, Actual: {client.ProtocolType}");
+
+            // Validate accessing end points
+            Assert.Equal(orig.LocalEndPoint, client.LocalEndPoint);
+            Assert.Equal(orig.RemoteEndPoint, client.RemoteEndPoint);
+
+            // Validating accessing other properties
+            Assert.Equal(orig.Available, client.Available);
+            Assert.True(orig.Blocking);
+            Assert.True(client.Blocking);
+            AssertEqualOrSameException(() => orig.DontFragment, () => client.DontFragment);
+            AssertEqualOrSameException(() => orig.EnableBroadcast, () => client.EnableBroadcast);
+            Assert.Equal(orig.ExclusiveAddressUse, client.ExclusiveAddressUse);
+            Assert.Equal(orig.Handle, client.Handle);
+            Assert.Equal(orig.IsBound, client.IsBound);
+            Assert.Equal(orig.LingerState.Enabled, client.LingerState.Enabled);
+            Assert.Equal(orig.LingerState.LingerTime, client.LingerState.LingerTime);
+            AssertEqualOrSameException(() => orig.MulticastLoopback, () => client.MulticastLoopback);
+            Assert.Equal(orig.NoDelay, client.NoDelay);
+            Assert.Equal(orig.ReceiveBufferSize, client.ReceiveBufferSize);
+            Assert.Equal(orig.ReceiveTimeout, client.ReceiveTimeout);
+            Assert.Equal(orig.SendBufferSize, client.SendBufferSize);
+            Assert.Equal(orig.SendTimeout, client.SendTimeout);
+            Assert.Equal(orig.Ttl, client.Ttl);
+            Assert.Equal(orig.UseOnlyOverlappedIO, client.UseOnlyOverlappedIO);
+
+            // Validate setting various properties on the new instance and seeing them roundtrip back to the original.
+            client.ReceiveTimeout = 42;
+            Assert.Equal(client.ReceiveTimeout, orig.ReceiveTimeout);
+
+            // Validate sending and receiving
+            Assert.Equal(1, await client.SendAsync(new byte[1] { 42 }, SocketFlags.None));
+            var buffer = new byte[1];
+            Assert.Equal(1, await server.ReceiveAsync(buffer, SocketFlags.None));
+            Assert.Equal(42, buffer[0]);
+
+            Assert.Equal(1, await server.SendAsync(new byte[1] { 42 }, SocketFlags.None));
+            buffer[0] = 0;
+            Assert.Equal(1, await client.ReceiveAsync(buffer, SocketFlags.None));
+            Assert.Equal(42, buffer[0]);
+        }
+
+        [PlatformSpecific(TestPlatforms.Windows | TestPlatforms.Linux)] // OSX/FreeBSD doesn't support SO_ACCEPTCONN, so we can't query for whether a socket is listening
```

Comment:
@wfurt, OSX has SO_ACCEPTCONN in its header file, but it errors with ENOPROTOOPT when trying to use it at the socket level, and the macOS and FreeBSD man page for getsockopt omit SO_ACCEPTCONN from the list of options supported for SOL_SOCKET.  Are you aware of an alternative?

Reply:
no, but I can take a look.

Reply:
Thanks. I'll merge this and we can follow-up if there's a better answer.